frontend/src/metabase/visualizations/echarts/cartesian/waterfall/model/dataset.ts

```diff
@@ -7,103 +7,69 @@ import {
 import { checkNumber } from "metabase/lib/types";
 
 import type { ComputedVisualizationSettings } from "metabase/visualizations/types";
-import type {
-  WaterfallDataset,
-  WaterfallDatum,
-  WaterfallEmptyValue,
-} from "../types";
-import { DATASET_DIMENSIONS } from "../constants";
-
-function createDatum({
```

Comment:
This function didn't provide much value over just adding defaults in `getWaterfallDataset` when creating datums, so I removed it.